api_app/auth.py

```diff
@@ -0,0 +1,128 @@
+import logging
+
+from django.contrib.auth import authenticate, login, logout
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.exceptions import APIException
+from rest_framework_simplejwt.tokens import RefreshToken
+from rest_framework_simplejwt.views import TokenRefreshView
+from rest_framework.decorators import (
+    api_view,
+    authentication_classes,
+    permission_classes,
+)
+
+from api_app import serializers
+
+
+logger = logging.getLogger(__name__)
+
+
+""" Auth API endpoints """
+
+
+@api_view(["POST"])
+@authentication_classes([])
+@permission_classes([])
+def obtain_user_token(request):
+    """
+    REST endpoint to obtain user auth token via authentication
+
+    :param username: string
+        username of registered user
+    :param password: string
+        password of registered user
+
+    :return 202:
+        if accepted
+    :return 404:
+        if failed
+    """
+    try:
+        username = request.data["username"]
+        logger.info(f"obtain_user_token received request for '{username}'.")
+        password = request.data["password"]
+        auth_user = authenticate(username=username, password=password)
+        if auth_user:
+            # ---- Todo ----
+            # should or should not login the admin ?
+            # verify with user-agent ?
+            logger.info(f"headers: {request.headers}")
+            if auth_user.is_superuser:
+                logger.info(f"administrator:'{username}' was logged in.")
+                login(request, auth_user)
+            # ---------------------------
+
+            logger.debug(f"obtain_user_token token created for '{username}'.")
+            refresh = RefreshToken.for_user(auth_user)
+            # adding custom 'username' claim
+            refresh["username"] = auth_user.username
+            return Response(
+                {"refresh": str(refresh), "access": str(refresh.access_token)},
+                status=status.HTTP_202_ACCEPTED,
+            )
+        raise APIException("No such user exists or Incorrect credentials")
+
+    except APIException as e:
+        logger.exception(f"obtain_user_token exception: {e}.")
+        return Response({"error": e}, status=status.HTTP_404_NOT_FOUND)
```

Comment:
if APIException triggers, then it would not be JSONSerializable and would trigger a 500 response. By casting the error as string maybe we solve this problem. In any case, we should add a more generic Exception clause to catch the unexpected. (However I noticed that the error ("No such user exists or Incorrect credentials") is not correctly shown by the client even with the previous adjustment)